session.go

```diff
@@ -563,8 +566,9 @@ runLoop:
 			// We do all the interesting stuff after the switch statement, so
 			// nothing to see here.
 		case <-s.sendingScheduled:
-			// We do all the interesting stuff after the switch statement, so
-			// nothing to see here.
+		// We do all the interesting stuff after the switch statement, so
+		// nothing to see here.
+		case <-sendQueueAvailable:
```

Comment:
Do we need to drain `sendQueueAvailable` if we end up taking another path in this switch, and then enter the if checks below? Otherwise it could be that we incorrectly assume there's space available even if there isn't, right?

Reply:
The `select` can unblock for multiple reasons (sending scheduled, receiving a packet, etc.), so we always have to check if there's room in the send queue before sending a packet: https://github.com/lucas-clemente/quic-go/blob/b4ffb97831546fc8336ed5f0c37ed6b6c8c40246/session.go#L621-L629

Reply:
I think I misunderstood your comment. Draining the channel makes sense if `WouldBlock()` returns `true`.

Reply:
Actually, that would be racy (the send queue could have unblocked between the calls to `WouldBlock` and Available`.